charms/kserve-controller/src/charm.py

```diff
@@ -76,6 +89,25 @@ def _context(self):
             "cert": f"'{ca_context.decode('utf-8')}'",
         }
 
+    @property
+    def _inference_service_context(self):
+        """Context for rendering the inferenceservive-config ConfigMap."""
+        # Ensure any input is valid for deployment mode
+        deployment_mode = self.model.config["deployment-mode"].lower()
+        if deployment_mode == "serverless":
+            deployment_mode = "Serverless"
+        deployment_mode = "RawDeployment"
```

Comment:
I do not know, if the user makes a typo?

what if we go into blocked state if the option is incorrect?
```
if serverless:
elif raw:
else: Blocked
```

Reply:
yeah, I can add something.